R/g.part1.R

```diff
@@ -392,8 +391,23 @@ g.part1 = function(datadir = c(), outputdir = c(), f0 = 1, f1 = c(),
   
   #--------------------------------------------------------------------------------
   # Run the code either parallel or in serial (file index starting with f0 and ending with f1)
- 
-  
+  cores = parallel::detectCores()
+  Ncores = cores[1]
+  if (Ncores > 3) {
+    
+    if (length(params_general[["maxNcores"]]) == 0) params_general[["maxNcores"]] = Ncores
+    Ncores2use = min(c(Ncores - 1, params_general[["maxNcores"]], (f1 - f0) + 1))
+    if (Ncores2use > 1) {
+      cl <- parallel::makeCluster(Ncores2use) # not to overload your computer
+      doParallel::registerDoParallel(cl)
+    } else {
+      # Don't process in parallel if only one core
+      params_general[["do.parallel"]] = FALSE
+    }
+  } else {
+    cat(paste0("\nparallel processing not possible because number of available cores (",Ncores,") < 4"))
```

Comment:
This warning is also printed if `do.parallel = FALSE`

Reply:
good point, I will edit now

Reply:
... just committed latest edits. I have a meeting now, but will test everything once more tomorrow and then prepare for submission to CRAN.

Reply:
These all good to me, Vincent.